core/src/main/scala/scalaz/EitherT.scala

```diff
@@ -287,6 +287,39 @@ object EitherT extends EitherTInstances {
 
 }
 
+object EitherTNondeterminism {
+  implicit def eitherTNondeterminism[F[_], L](implicit F0: Nondeterminism[F]): Nondeterminism[EitherT[F, L, ?]] =
+    new Nondeterminism[EitherT[F, L, ?]] {
```

Comment:
This instance should be integrated into the existing tapestry of instances, rather than living in its own universe.

1. Add `private trait EitherTNondeterminism[F[_], E] extends Nondeterminism[EitherT[F, E, ?]] with EitherTMonad[F, E]`. This will inherit all `Monad` implementations and any overrides done elsewhere for efficiency, so even when someone gets your new instance, they'll still have the most efficient implementations.  For example, you get the `map` override for free.
2. Your trait should have only one abstract member, `implicit def F: Nondeterminism[F]`.
3. Here is the tricky part: the implicit must go somewhere in the tower of `EitherTInstances*` `sealed abstract class`es. It must go somewhere that doesn't cause ambiguity. We have "if it compiles it works" tests in `EitherTTest.scala` to help you avoid regressions in this area; you'll need to *extend* this set to incorporate your new instance as well, ensuring that it resolves unambiguously without importing.
4. That concrete implementation only sets `implicit def F`; see every other instance for examples.